misc/formatter/modules/formatter-cli/src/main/java/org/ballerinalang/formatter/cli/FormatUtil.java

```diff
@@ -0,0 +1,358 @@
+        emptyPrintStream = new EmptyPrintStream();
+        CompilerContext context = getCompilerContext(sourceRoot);
+        Compiler compiler = Compiler.getInstance(context);
+        // Set an EmptyPrintStream to hide unnecessary outputs from compiler.
+        compiler.setOutStream(emptyPrintStream);
+        return compiler.compile(moduleName);
+    }
+
+    private static void formatAndWrite(BLangCompilationUnit compilationUnit, Path sourceRootPath,
+                                       List<String> formattedFiles, boolean dryRun) throws IOException {
+        String fileName = sourceRootPath.toString() + File.separator
+                + "src"
+                + File.separator
+                + compilationUnit.getPosition().getSource().getPackageName()
+                + File.separator
+                + compilationUnit.getPosition().getSource().getCompilationUnitName();
+
+        // Format and get the formatted source.
+        String formattedSource = Formatter.format(new String(Files.readAllBytes(Paths.get(fileName)),
+                StandardCharsets.UTF_8));
+
+        if (doChangesAvailable(formattedSource, formattedSource)) {
+            if (!dryRun) {
+                // Write formatted content to the file.
+                FormatUtil.writeFile(fileName, formattedSource);
+            }
+            formattedFiles.add(fileName);
+        }
+    }
+
+    private static List<String> iterateAndFormat(BLangPackage bLangPackage, Path sourceRootPath, boolean dryRun)
+            throws IOException {
+        List<String> formattedFiles = new ArrayList<>();
+
+        // Iterate compilation units and format.
+        for (BLangCompilationUnit compilationUnit : bLangPackage.getCompilationUnits()) {
+            formatAndWrite(compilationUnit, sourceRootPath, formattedFiles, dryRun);
+        }
+
+        // Iterate testable packages and format.
+        for (BLangTestablePackage testablePackage : bLangPackage.getTestablePkgs()) {
+            for (BLangCompilationUnit compilationUnit : testablePackage.getCompilationUnits()) {
+                formatAndWrite(compilationUnit, sourceRootPath, formattedFiles, dryRun);
+            }
+        }
+
+        return formattedFiles;
+    }
+
+    /**
+     * Get prepared compiler context.
+     *
+     * @param sourceRootPath ballerina compilable source root path
+     * @return {@link CompilerContext} compiler context
+     */
+    private static CompilerContext getCompilerContext(Path sourceRootPath) {
+        CompilerPhase compilerPhase = CompilerPhase.DEFINE;
+        CompilerContext context = new CompilerContext();
+        CompilerOptions options = CompilerOptions.getInstance(context);
+        options.put(PROJECT_DIR, sourceRootPath.toString());
+        options.put(OFFLINE, Boolean.toString(false));
+        options.put(COMPILER_PHASE, compilerPhase.toString());
+        options.put(SKIP_TESTS, Boolean.toString(false));
+        options.put(TEST_ENABLED, "true");
```

Comment:
Should tests run ?

Reply:
Since this is for the formatting purpose, running tests is not required.